File: 0x06-python-more_classes/3-rectangle.py
#!/usr/bin/python3
"""
Program that defines a Rectangle
"""


class Rectangle:
    """
    Defines methods to create attributes.
    """

    def __init__(self, width=0, height=0):
        """Init the object's atribute"""
        self.width = width
        self.height = height

    @property
    def width(self):
        """Get the width"""
        return self.__width

    @width.setter
    def width(self, value):
        """Set the width and raise"""
        if type(value) != int:
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @property
    def height(self):
        """Get the height"""
        return self.__height

    @height.setter
    def height(self, value):
        """Set the height and raise"""
        if type(value) != int:
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    def area(self):
        """Returns the area of the rectangle"""
        return self.__width * self.__height

    def perimeter(self):
        """Returns the perimeter of the rectangle"""
        if self.__width == 0 or self.__height == 0:
            return 0
        else:
            return (self.__width + self.__height) * 2

    def __str__(self):
        """Prints the rectangle with str magic method"""
        if self.__width == 0 or self.__height == 0:
            return ""
        for i in range(self.__height - 1):
            print('#' * self.__width)
        print('#' * self.__width, end='')
        return ""

File: 0x01-javascript_objects_scopes_closures/3-rectangle.js
#!/usr/bin/node
// Write a class Rectangle that
module.exports = class Rectangle {
    constructor (width, height) {
      if (typeof width === 'number' && typeof height === 'number' && width > 0 && height > 0) {
        this.width = width;
        this.height = height;
      }
    }
    print() {
        if (!this.width || !this.height) {
          console.log('Empty rectangle');
        } else {
          let row = '';
          for (let i = 0; i < this.width; i++) {
            row += 'X';
          }
          for (let i = 0; i < this.height; i++) {
            console.log(row);
          }
        }
      }
  }
